src/plugin.js

```diff
@@ -66,6 +69,19 @@ const contribAdsPlugin = function(options) {
   // Set up redispatching of player events
   player.on(videoEvents, redispatch);
 
+  // Set up features to block content playback while waiting for ads.
+  // Play middleware is only supported on later versions of video.js
+  // and on desktop currently(as the user-gesture requirement on mobile
+  // will disallow calling play once play blocking is lifted)
+  // The middleware must also be registered outside of the plugin,
+  // to avoid a middleware factory being created for each player
+  if (isMiddlewareMediatorSupported() && settings.debug) {
+    videojs.log('ADS:', 'Play middleware has been registered with videojs');
```

Comment:
It would be a little tidier to move this to where middleware is actually registered.